src/System.Security.Cryptography.Pkcs/ref/System.Security.Cryptography.Pkcs.netcoreapp.cs

```diff
@@ -5,8 +5,72 @@
 // Changes to this file must follow the http://aka.ms/api-review process.
 // ------------------------------------------------------------------------------
 
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
 namespace System.Security.Cryptography.Pkcs
 {
+    public enum Rfc3161RequestResponseStatus
+    {
+        Unknown = 0,
+        Accepted = 1,
+        DoesNotParse = 2,
+        RequestFailed = 3,
+        HashMismatch = 4,
+        VersionTooNew = 5,
+        NonceMismatch = 6,
+        RequestedCertificatesMissing = 7,
+        UnexpectedCertificates = 8,
+    }
+    public sealed partial class Rfc3161TimestampRequest
+    {
+        private Rfc3161TimestampRequest() { }
+        public int Version => throw null;
+        public ReadOnlyMemory<byte> GetMessageHash() => throw null;
+        public Oid HashAlgorithmId => throw null;
+        public Oid RequestedPolicyId => throw null;
+        public bool RequestSignerCertificate => throw null;
+        public ReadOnlyMemory<byte>? GetNonce() => throw null;
+        public bool HasExtensions => throw null;
+        public X509ExtensionCollection GetExtensions() => throw null;
+        public Task<Rfc3161TimestampToken> SubmitRequestAsync(Uri uri, TimeSpan timeout) => throw null;
+        public byte[] Encode() => throw null;
+        public bool TryEncode(Span<byte> destination, out int bytesWritten) => throw null;
+        public Rfc3161TimestampToken AcceptResponse(ReadOnlyMemory<byte> source, out int bytesRead) => throw null;
+        public bool TryAcceptResponse(ReadOnlyMemory<byte> source, out int bytesRead, out Rfc3161RequestResponseStatus status, out Rfc3161TimestampToken token) => throw null;
+        public static Rfc3161TimestampRequest BuildForData(ReadOnlySpan<byte> data, HashAlgorithmName hashAlgorithm, Oid requestedPolicyId = null, ReadOnlyMemory<byte>? nonce = null, bool requestSignerCertificates = false, X509ExtensionCollection extensions = null) => throw null;
+        public static Rfc3161TimestampRequest BuildForHash(ReadOnlyMemory<byte> hash, HashAlgorithmName hashAlgorithm, Oid requestedPolicyId = null, ReadOnlyMemory<byte>? nonce = null, bool requestSignerCertificates = false, X509ExtensionCollection extensions = null) => throw null;
```

Comment:
Why does BuildForData take a `ReadOnlySpan` but BuildForHash takes a `ReadOnlyMemory`?

Reply:
A ROMemory can be serialized, an ROSpan can't.  So the Data one (which hashes) has to allocate, the Hash one expects you to allocate for me.

Reply:
The pattern considered here forces the caller to ensure the memory pointed to by the `ReadOnlyMemory` instance isn't deallocated or mutated after the `Rfc3161TimestampRequest` instance is created. I'm concerned that this might make the API more difficult to use since it forces the caller to consider object lifetimes. Would this API be easier to use if we made a copy of the data internally? I know it's an extra allocation, but I can't imagine this particular API is ever going to end up in a hot path anywhere.

Reply:
Once serialized, a copy is made.  The caller only has to have their data invariant during the execution of this method.  I can make use of ArrayPool if that isn't feeling obvious.

Reply:
Ah, I got confused and thought this returned a `Rfc3161TimestampReq` struct, not a `Rfc3161TimestampRequest` class. Ignore previous comment. :)